# Alejando

<hr style="border:2px solid gray">

In [1]:
#imports
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [2]:
def calculate_final_grade(exercise_grades, project_grades, attendance_grade, quiz_grades):
    # Define the weights for each category
    exercise_weight = 0.2
    project_weight = 0.4
    attendance_weight = 0.1
    quiz_weight = 0.3

    # Calculate the overall weighted score for each category
    exercise_score = sum(exercise_grades) / len(exercise_grades)
    project_score = sum(project_grades) / len(project_grades)
    quiz_score = sum(quiz_grades) / len(quiz_grades)
    
    # Calculate the overall weighted score
    final_grade = (exercise_score * exercise_weight) + (project_score * project_weight) + \
                             (attendance_grade * attendance_weight) + (quiz_score * quiz_weight)

    return final_grade


# Input the existing grades for the student
exercise_grades = [100, 96, 91,95,0,51,24,80,40,33,50] 
project_grades = [93, 57, 80, 21, 0] 
attendance_grade = 82  
quiz_grades = [67,37,0]  

# Calculate the final grade
final_grade = calculate_final_grade(exercise_grades, project_grades, attendance_grade, quiz_grades)

# Print the final grade
print("The final grade for the student is:", final_grade)

The final grade for the student is: 50.68


<hr style="border:2px solid gray">

## Evaluation of Alejandro's Performance to Date

To this point, Quintela has recieved grades for a total of:
- 11 exercises (19 remaining)
- 5 projects (5 remaining)
- 3 quizzes (2 remaining)
- 1 attendance grade

In [3]:
exercise_weight = 0.2
project_weight = 0.4
attendance_weight = 0.1
quiz_weight = 0.3

<b>Current Overall Exercise Grade</b> (Out of 100)

In [4]:
exercise_score = sum(exercise_grades) / len(exercise_grades)
exercise_score

60.0

<b>Current Overall Project Grade</b> (Out of 100)

In [5]:
project_score = sum(project_grades) / len(project_grades)
project_score

50.2

<b>Current Overall Quize Grade</b> (Out of 100)

In [6]:
quiz_score = sum(quiz_grades) / len(quiz_grades)
quiz_score

34.666666666666664

<b>Overall Final Grade</b> (Out of 100)

In [7]:
overall_weighted_score = (exercise_score * exercise_weight) + (project_score * project_weight) + \
                             (attendance_grade * attendance_weight) + (quiz_score * quiz_weight)
overall_weighted_score

50.68

### Takeaways

If the course were to end today, Alejandro would finish with a 50.68% grade. 
- Exercise grade: 60/100
- Project Grade: 50.2/100
- Quiz Grade: 34.66/100
- Attendance: 82/100

<hr style="border:2px solid gray">

## Projected through Regression Progress Report

We can project that through Regression Progress Report, Quintela will have a total of:
- 18 exercises (7 more than current)
- 7 projects (2 more than current)
- 5 quizzes (no additional)
- 1 attendance grade

In [11]:
def calculate_final_grade(exercise_grades, project_grades, attendance_grade, quiz_grades):
    # Define the weights for each category
    exercise_weight = 0.2
    project_weight = 0.4
    attendance_weight = 0.1
    quiz_weight = 0.3

    # Calculate the overall weighted score for each category
    exercise_score = sum(exercise_grades) / len(exercise_grades)
    project_score = sum(project_grades) / len(project_grades)
    quiz_score = sum(quiz_grades) / len(quiz_grades)

    # Calculate the overall weighted score
    through_regression_grade = (exercise_score * exercise_weight) + (project_score * project_weight) + \
                             (attendance_grade * attendance_weight) + (quiz_score * quiz_weight)

    return through_regression_grade


# Input the existing grades for the student
exercise_grades = [100, 96, 91,95,0,51,24,80,40,33,50,100,100,100,100,100,100,100] 
project_grades = [93, 57, 80, 21, 0, 100, 100] 
attendance_grade = 82  
quiz_grades = [67,37,0, 100, 100]  

# Calculate the final grade
through_regression_grade = calculate_final_grade(exercise_grades, project_grades, attendance_grade, quiz_grades)

# Print the final grade
print("The final grade for the student is:", through_regression_grade)

The final grade for the student is: 67.32253968253968


In [9]:
through_regression_exercise_score = sum(exercise_grades) / len(exercise_grades)
through_regression_project_score = sum(project_grades) / len(project_grades)
through_regression_quiz_score = sum(quiz_grades) / len(quiz_grades)
through_regression_exercise_score, through_regression_project_score, through_regression_quiz_score

(74.11764705882354, 64.42857142857143, 60.8)

If we projected all assignments until the regression progress report to be 100%, Alejandro's grade at the end of the Regression progress report will be 67.32%. Unfortunetly, he is already on Academic Probation, so he will be removed from the program.

- Exercise grade: 74.11/100
- Project Grade: 64.42/100
- Quiz Grade: 60.8/100
- Attendance: 82/100

<hr style="border:2px solid gray">

## Projected Through End of Course

We can project that Quintela will have a total of:
- 30 exercises
- 10 projects
- 5 quizzes
- 1 attendance grade

This is a simulation to show a grade of 100% on every single assignment (exercise, quiz, and project) until the end of the course.

In [ ]:
def calculate_final_grade(exercise_grades, project_grades, attendance_grade, quiz_grades):
    # Define the weights for each category
    exercise_weight = 0.2
    project_weight = 0.4
    attendance_weight = 0.1
    quiz_weight = 0.3

    # Calculate the overall weighted score for each category
    exercise_score = sum(exercise_grades) / len(exercise_grades)
    project_score = sum(project_grades) / len(project_grades)
    quiz_score = sum(quiz_grades) / len(quiz_grades)

    # Calculate the overall weighted score
    possible_final_grade = (exercise_score * exercise_weight) + (project_score * project_weight) + \
                             (attendance_grade * attendance_weight) + (quiz_score * quiz_weight)

    return possible_final_grade


# Input the existing grades for the student
exercise_grades = [100, 96, 91,95,0,51,24,80,40,33,50,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100] 
project_grades = [93, 57, 80, 21, 0, 100, 100, 100, 100, 100] 
attendance_grade = 82  
quiz_grades = [67,37,0, 100, 100]  

# Calculate the final grade
possible_final_grade = calculate_final_grade(exercise_grades, project_grades, attendance_grade, quiz_grades)

# Print the final grade
print("The final grade for the student is:", possible_final_grade)

In [ ]:
possible_exercise_score = sum(exercise_grades) / len(exercise_grades)
possible_project_score = sum(project_grades) / len(project_grades)
possible_quiz_score = sum(quiz_grades) / len(quiz_grades)
possible_exercise_score, possible_project_score, possible_quiz_score

If every single grade were projected to be 100%, Alejandro could technically pass with a 73.55%. 

- Exercise grade: 85.3/100
- Project Grade: 75.1/100
- Quiz Grade: 60.8/100
- Attendance: 82/100

There is literally no room here. If he gets anything other than a 100% on any of his assignments (project or quiz), he will drop well below 70% and fail the program.